model/build/build.go

```diff
@@ -317,3 +331,78 @@ func (b Builds) InsertMany(ctx context.Context, ordered bool) error {
 	_, err := evergreen.GetEnvironment().DB().Collection(Collection).InsertMany(ctx, b.getPayload(), &options.InsertManyOptions{Ordered: &ordered})
 	return errors.Wrap(err, "bulk inserting builds")
 }
+
+// GetFinishedNotificationDescription returns a description of successful/failed tasks for the build,
+// to be used by jobs and notification processing.
+func (b *Build) GetFinishedNotificationDescription(tasks []task.Task) string {
+	success := 0
+	failed := 0
+	systemError := 0
+	other := 0
+	noReport := 0
+	for _, t := range tasks {
+		switch {
+		case t.Status == evergreen.TaskSucceeded:
+			success++
+
+		case t.Status == evergreen.TaskFailed:
+			failed++
+
+		case statusIsSystemFailure(t.Status):
+			systemError++
+
+		case utility.StringSliceContains(evergreen.TaskUncompletedStatuses, t.Status):
+			noReport++
+
+		default:
+			other++
+		}
+	}
+
+	grip.ErrorWhen(other > 0, message.Fields{
+		"source":   "status updates",
+		"message":  "unknown task status",
+		"build_id": b.Id,
+	})
+
+	if success == 0 && failed == 0 && systemError == 0 && other == 0 {
+		return "no tasks were run"
+	}
+
+	desc := fmt.Sprintf("%s, %s", taskStatusSubformat(success, "succeeded"),
+		taskStatusSubformat(failed, "failed"))
+	if systemError > 0 {
+		desc += fmt.Sprintf(", %d internal errors", systemError)
+	}
+	if other > 0 {
+		desc += fmt.Sprintf(", %d other", other)
+	}
+
+	return b.appendTime(desc)
+}
+
+func statusIsSystemFailure(status string) bool {
```

Comment:
Can we instead model this function like `IsSystemFailedTaskStatus` and keep it in globals?

Reply:
having wrote half this code a month ago i have no idea why i even did this haha, since this whole function is just TaskStatusToDesc moved except for this change. switched to use the existing IsSystemFailedTaskStatus.